packages/core/src/app/customer/Customer.tsx

```diff
@@ -433,16 +475,26 @@ export function mapToWithCheckoutCustomerProps(
     { checkoutService, checkoutState }: CheckoutContextProps
 ): WithCheckoutCustomerProps | null {
     const {
-        data: { getBillingAddress, getCustomerAccountFields, getCheckout, getCustomer, getSignInEmail, getConfig },
+        data: { getBillingAddress, getCustomerAccountFields, getCheckout, getCustomer, getSignInEmail, getConfig, getPaymentMethod, getCart },
         errors: { getSignInError, getSignInEmailError, getCreateCustomerAccountError },
         statuses: { isContinuingAsGuest, isExecutingPaymentMethodCheckout, isInitializingCustomer, isSigningIn, isSendingSignInEmail, isCreatingCustomerAccount },
     } = checkoutState;
 
+    const cart = getCart();
     const billingAddress = getBillingAddress();
     const checkout = getCheckout();
     const customer = getCustomer();
     const signInEmail = getSignInEmail();
     const config = getConfig();
+    let stripeUpeLinkEnabled = false;
+
+    if (cart) {
+        const stripeUpe = getPaymentMethod('card', PaymentMethodId.StripeUPE);
+        const linkEnabled = stripeUpe?.initializationData.enableLink || false;
+        const stripeUpeSupportedCurrency = cart?.currency.code === 'USD' || false;
```

Comment:
since this block is inside if (cart), the optional chaining is not required
```suggestion
        const stripeUpeSupportedCurrency = cart.currency.code === 'USD' || false;
```